src/util.js

```diff
@@ -87,3 +87,33 @@ const nowAsString = () => {
     + pad2(now.getHours()) + '-'
     + pad2(now.getMinutes());
 };
+
+// Decodes the given hex string into a new string. If the given string contains
+// characters that are not hexadecimal digits or if the string's length is odd,
+// this function will throw an exception.
+export const decodeHexString = hex => {
```

Comment:
I don't think this will handle multi-byte encodings well (though I didn't spent a lot of time going through the code either). But why not just use Base64 encoding? JavaScript does have the `btoa('...');` method, doesn't it?

Reply:
Base64 still uses a lot of characters that could be considered "special". Hex encoding, while way less space efficient, uses only the most basic characters. So I thought: if we already add this feature for systems where URLs get corrupted, we might as well go all the way.